includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1075,13 +1089,48 @@ public function generate_account_statement_descriptor_html( $key, $data ) {
 	 */
 	public function get_option( $key, $empty_value = null ) {
 		switch ( $key ) {
+			case 'enabled':
+				return parent::get_option( static::METHOD_ENABLED_KEY, $empty_value );
 			case 'account_statement_descriptor':
 				return $this->get_account_statement_descriptor();
 			default:
 				return parent::get_option( $key, $empty_value );
 		}
 	}
 
+	/**
```

Comment:
I had the impression that the new method introduced in order to make `\WC_Payment_Gateway_WCPay::is_enabled`, `\WC_Payment_Gateway_WCPay::disable`, `\WC_Payment_Gateway_WCPay::enable` for the new gateways.

I suggest simplifying this:
- introduce the setting key as a constant (no dynamic calculation)
- refactor the methods to use the constant
- override the constant for each gateway

That would make this PR even more awesome =)

Reply:
I think I'll let the setting squad handle further iterations around this. For now I want the main WC_Payment_Gateway_WCPay gateway to be responsible for managing all setting so they will stay in the same option key and not need migrating. There will be payment gateway specific setting but for now all should use the same. Eventually I think we want an option key for common setting and an option key for each gateway's settings.

Reply:
Sounds good.